pkg/config/kube/conversion.go

```diff
@@ -59,9 +70,15 @@ func ConvertProtocol(name string, proto coreV1.Protocol) protocol.Instance {
 			name = name[:i]
 		}
 		p := protocol.Parse(name)
-		if p != protocol.UDP && p != protocol.Unsupported {
+		if p != protocol.UDP {
 			out = p
 		}
 	}
+
+	// Make TCP as default protocol for well know ports if protocol is not specified.
+	if _, has := wellKnownPorts[port]; has && out == protocol.Unsupported {
```

Comment:
This seems like a separate change. We should probably pull this out into its own PR with its own testing.

Is this something that this PR relies on?

Reply:
It relies on protocol sniffing. We won't output protocol.Unsupported if protocol sniffing is disabled.

Reply:
@crazyxy I'm not sure that's true. Nowhere in this function do I see a check for whether protocol sniffing is enabled. It seems that it applies regardless, which is my concern.

Reply:
Cannot check protocol sniffing here as it relies on proxy instance. The output is different before and after the change and that's why I add the check ModelProtocolToListenerProtocol.